src/pydocstyle/cli.py

```diff
@@ -49,7 +49,8 @@ def run_pydocstyle():
 
     count = 0
     for error in errors:
-        sys.stdout.write('%s\n' % error)
+        if hasattr(error, 'code'):
```

Comment:
I'd check that the type of `error` does not inherit from `BaseException`. Ducktyping here seems... odd (imo).